config/src/main/java/org/springframework/security/config/http/HttpConfigurationBuilder.java

```diff
@@ -387,7 +387,8 @@ else if (StringUtils.hasText(sessionAuthStratRef)) {
 			concurrentSessionStrategy = BeanDefinitionBuilder
 					.rootBeanDefinition(ConcurrentSessionControlAuthenticationStrategy.class);
 			concurrentSessionStrategy.addConstructorArgValue(this.sessionRegistryRef);
-			String maxSessions = sessionCtrlElt.getAttribute("max-sessions");
+			String maxSessions = this.pc.getReaderContext().getEnvironment()
```

Comment:
Would you please add a test similar to [this one](https://github.com/spring-projects/spring-security/blob/2abf59b695b3ad14719299ed17ff47b181eed802/config/src/test/java/org/springframework/security/config/http/SessionManagementConfigTests.java#L348) in order to demonstrate that the placeholder works?

Tests like [this one](https://github.com/spring-projects/spring-security/blob/2abf59b695b3ad14719299ed17ff47b181eed802/config/src/test/java/org/springframework/security/config/http/PlaceHolderAndELConfigTests.java#L55) may prove useful for getting the property set.